presto-main/src/test/java/com/facebook/presto/cost/TestFilterStatsRule.java

```diff
@@ -0,0 +1,330 @@
+                                .setLowValue(10)
+                                .setHighValue(15)
+                                .setDistinctValuesCount(4)
+                                .setNullsFraction(0.1)
+                                .build())
+                        .build())
+                .check(check -> check
+                        .outputRowsCount(2)
+                        .symbolStats("i1", assertion -> assertion
+                                .lowValue(5)
+                                .highValue(5)
+                                .distinctValuesCount(1)
+                                .dataSizeUnknown()
+                                .nullsFraction(0))
+                        .symbolStats("i2", assertion -> assertion
+                                .lowValue(0)
+                                .highValue(3)
+                                .dataSizeUnknown()
+                                .distinctValuesCount(2)
+                                .nullsFraction(0))
+                        .symbolStats("i3", assertion -> assertion
+                                .lowValue(10)
+                                .highValue(15)
+                                .dataSizeUnknown()
+                                .distinctValuesCount(1.9)
+                                .nullsFraction(0.05)));
+    }
+
+    @Test
+    public void testNoNdvsForEqualityFilter()
+    {
+        // no ndvs, so can't estimate equality predicate
+        tester().assertStatsFor(pb -> pb
+                .filter(expression("i1 = 5"),
+                        pb.values(pb.symbol("i1"), pb.symbol("i2"), pb.symbol("i3"))))
+                .withSourceStats(0, PlanNodeStatsEstimate.builder()
+                        .setOutputRowCount(10)
+                        .addSymbolStatistics(new Symbol("i1"), SymbolStatsEstimate.builder()
+                                .setLowValue(1)
+                                .setHighValue(10)
+                                .setNullsFraction(0)
+                                .build())
+                        .addSymbolStatistics(new Symbol("i2"), SymbolStatsEstimate.builder()
+                                .setLowValue(0)
+                                .setHighValue(3)
+                                .setDistinctValuesCount(4)
+                                .setNullsFraction(0)
+                                .build())
+                        .addSymbolStatistics(new Symbol("i3"), SymbolStatsEstimate.builder()
+                                .setLowValue(10)
+                                .setHighValue(15)
+                                .setDistinctValuesCount(4)
+                                .setNullsFraction(0.1)
+                                .build())
+                        .build())
+                .check(check -> check
+                        .outputRowsCountUnknown()
+                        .symbolStats("i1", assertion -> assertion
```

Comment:
This implementation may change, so we may no longer want to return any symbol stats if output rows count is uknown. I would remove the symbol stats checks here.